agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/DiseaseRESTInterface.java

```diff
@@ -134,9 +187,35 @@
             @ApiParam(name = "id", value = "Disease by DOID: e.g. DOID:9952", required = true, type = "String")
             @PathParam("id") String id,
             @ApiParam(name = "limit", value = "Number of rows returned", defaultValue = "20")
-            @DefaultValue("20") @QueryParam("limit") int limit,
+            @DefaultValue("20") @QueryParam("limit") Integer limit,
             @ApiParam(name = "page", value = "Page number")
-            @DefaultValue("1") @QueryParam("page") int page,
+            @DefaultValue("1") @QueryParam("page") Integer page,
+            @ApiParam(value = "Field / column name by which to sort", allowableValues = "Default,Gene,Disease,Species", defaultValue = "geneName")
+            @QueryParam("sortBy") String sortBy,
+            @ApiParam(value = "filter by model name")
+            @QueryParam("filter.modelName") String modelName,
+            @ApiParam(value = "filter by gene symbol")
+            @QueryParam("filter.geneName") String geneSymbol,
+            @ApiParam(value = "filter by species")
+            @QueryParam("filter.species") String species,
+            @ApiParam(value = "filter by disease")
+            @QueryParam("filter.disease") String disease,
+            @ApiParam(value = "filter by source")
+            @QueryParam("filter.source") String source,
+            @ApiParam(value = "filter by reference")
+            @QueryParam("filter.reference") String reference,
+            @ApiParam(value = "filter by evidence code")
+            @QueryParam("filter.evidenceCode") String evidenceCode,
+            @ApiParam(value = "ascending order: true or false", allowableValues = "true,false", defaultValue = "true")
+            @QueryParam("asc") String asc);
+
```

Comment:
Just wondering if maybe a better way to handle this instead of having many parameters we could do something like
List<String> filter_fields and List<String> filter_values and the URL string would look like:
`/endpoint?filter_fields=[disease,geneName]&filter_values=[diabetes,pax6]`

Reply:
Then we could just have a util that takes both those lists and returns a map that is ready to go for filtering. That was we could just put those two params on all endpoints to get filtering. I know we would loose the documentability of each field, but would gain not having to change things if we wanted to have a new way to filter.

Reply:
I know and I have been hoping for something like this but then Swagger will not show each individual parameter.

Reply:
It's just more in your face, i.e. the parameters are just more explicit.